### ISSUE TITLE
Track duplicates in DataAPI.refpool issue

### ISSUE THREAD

**Issue by bkamins** [JuliaData/DataFrames.jl#2558]
Follow up of https://github.com/JuliaData/DataFrames.jl/pull/2442#discussion_r528367569.

We should resolve it before 1.0 release as I understand that if we have duplicates in `DataAPI.refpool` we are going to get wrong results in split-apply-combine.

**Comment by bkamins** [comment]
CC @nalimilan @quinnj

**Comment by nalimilan** [comment]
We can fix it at any point after 1.0. Currently results are correct, we just have a check that is sometimes costly and not necessary for `PooledArray` and `CagegoricalArray`.

**Comment by bkamins** [comment]
Ah - so we have this check now? I thought we do not check uniqueness.

**Comment by bkamins** [comment]
This might be one of the reasons of slow H2O tests then for very large number of groups.

**Comment by nalimilan** [comment]
I don't think so: with 100 rows per group (instead of 5 in my example) the overhead turns out to be much smaller. I was surprised by this but it makes sense I guess. Anyway, saving a few percents wouldn't be bad.

**Comment by nalimilan** [comment]
The check is:
https://github.com/JuliaData/DataFrames.jl/blob/8645651e30785fbedca82a9f125d21d1f27a726e/src/dataframerow/utils.jl#L199
